python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -0,0 +1,234 @@
+                self.nlihcid = project[columns.index(column_name)]
+
+    def findRailStations(self, railStations,project,radiusinmeters,distCsvWriter):
+        """Finds all the rail stations within a given distance from a given project.  Writes to the given CSV file.
+
+        Parameters:
+        railStations - json object containing all the wmata rail station information
+        project - housing project object
+        radiusinmeters - radius in meteres
+        distCsvWriter - csvWriter for distance
+        mapbox_api_key - api key for mapbox REST services
+        """
+
+        lat = self.lat
+        lon = self.lon
+        Nlihc_id = self.nlihcid
+
+        for station in railStations:
+            walkDist = self.getWalkingDistance(lat, lon, str(station['Lat']), str(station['Lon']))
+
+            if walkDist <=radiusinmeters:
+                #railData.append([Nlihc_id, 'rail', station['Code'], "{0:.2f}".format(self.miles)])
+                distCsvWriter.writerow([Nlihc_id, 'rail', station['Code'], "{0:.2f}".format(self.miles)])
+
+        #self.result_to_csv(distHeader,railData,self.distOutputFileName)
+
+    def findBusStations(self, project,radiusinmeters, distCsvWriter):
+        lat = self.lat
+        lon = self.lon
+        Nlihc_id = self.nlihcid
+
+        wmata_headers = self.getWmataHeaders()
+
+        params = {'Lat': lat,
+                  'Lon' : lon,
+                  'Radius':str(radiusinmeters)}
+        response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', params=params, headers=wmata_headers)
+        data = response.json()
+
+        for stop in data['Stops']:
+            walkDist = self.getWalkingDistance(lat, lon, str(stop['Lat']), str(stop['Lon']))
+            if walkDist <= radiusinmeters: #within 0.5 miles walking
+                distCsvWriter.writerow([Nlihc_id, 'bus', stop['StopID'], "{0:.2f}".format(self.miles)])
+
+    def writeRailInfo(self, infoCsvWriter):
+        """Writes all rail station data to a given CSV writer. Returns the railStations json for future processing
+
+           Parameters:
+           infoCsvWriter - csv writer
+           wmata_api_key - api key for wmata REST services
+           """
+        print("Writing RAIL INFO")
+
+        wmata_headers = self.getWmataHeaders()
+
+        railResponse = requests.get("https://api.wmata.com/Rail.svc/json/jStations", headers=wmata_headers)
+        railStations = railResponse.json()['Stations']
+
+        for station in railStations:
+            #delimit list of lines with colon
+            lines = station["LineCode1"] #there is always at least one station
+            for line_code in ["LineCode2", "LineCode3", "LineCode4"]:
+                if station[line_code] != None:
+                    lines += ":" + station[line_code]
+            infoCsvWriter.writerow((station['Code'], 'rail',station['Name'],str(station['Lat']), str(station['Lon']),lines))
+
+        return railStations
```

Comment:
they both return json now. I want to also pass the bus json to _get_all_bus_stations so that I don't have to do the query a second time, but it has slightly different parameters, so I'm just going to make sure it doesn't make a huge difference first (on my To Do list)